src/poetry/installation/installer.py

```diff
@@ -337,8 +337,8 @@ def _do_install(self, local_repo: Repository) -> int:
         # Execute operations
         return self._execute(ops)
 
-    def _write_lock_file(self, repo: Repository, force: bool = True) -> None:
-        if force or (self._update and self._write_lock):
+    def _write_lock_file(self, repo: Repository, force: bool = False) -> None:
+        if self._write_lock and (force or self._update):
```

Comment:
This was a suggestion from @radoering [here](https://github.com/python-poetry/poetry/pull/3767/files#r842891790), and looking at the usage of the method, it does seem that the only place where we use `force` leaves `_write_lock` to `True`, so this seems to make sense.